tools/build_pgo.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -x
+
+merged_path=$1
+
+python -c 'import sys;assert sys.platform == "win32"'
+is_win=$?
+
+set -e
+# Create venv for instrumented build and test
+python -m venv build_pgo
+
+if [[ $is_win -eq 0 ]]; then
+    source build_pgo/Scripts/activate
+else
+    source build_pgo/bin/activate
+fi
+
+# Build with instrumentation
+pip install --prefer-binary -c constraints.txt -r requirements.txt setuptools-rust wheel
+RUSTFLAGS="-Cprofile-generate=/tmp/pgo-data" pip install --prefer-binary -c constraints.txt -e .
+RUSTFLAGS="-Cprofile-generate=/tmp/pgo-data" python setup.py build_rust --release --inplace
+pip install -c constraints.txt --prefer-binary -r requirements-dev.txt
```

Comment:
I actually did an editable install so the test pathing worked out correctly because it's using everything in place from the repo root. TBH I think it probably would "just work" if I removed the install and just did `build_rust` because it's all in the CWD (I'm not actually proposing we do this). I think we can combine the install lines, into a single call I agree it's probably not an issue, but it doesn't hurt to be more precise here. I would leave the `RUSTFLAGS` set on the `pip` call, while I don't think it's strictly necessary it doesn't hurt.

Reply:
```suggestion
pip install  -c constraints.txt setuptools-rust wheel setuptools
RUSTFLAGS="-Cprofile-generate=/tmp/pgo-data" pip install --prefer-binary -c constraints.txt -r requirements-dev.txt -e .
RUSTFLAGS="-Cprofile-generate=/tmp/pgo-data" python setup.py build_rust --release --inplace
```